src/rdkafka_broker.c

```diff
@@ -3614,20 +3650,125 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
                                 int32_t MessageSetSize;
                         } hdr;
                         rd_kafka_resp_err_t err;
+                        int64_t end_offset;
 
 			rd_kafka_buf_read_i32(rkbuf, &hdr.Partition);
 			rd_kafka_buf_read_i16(rkbuf, &hdr.ErrorCode);
 			rd_kafka_buf_read_i64(rkbuf, &hdr.HighwaterMarkOffset);
 
+                        end_offset = hdr.HighwaterMarkOffset;
+
                         if (rd_kafka_buf_ApiVersion(request) == 4) {
-                                int32_t AbortedTxCnt;
+                                int32_t AbortedTxnCnt;
                                 rd_kafka_buf_read_i64(rkbuf,
                                                       &hdr.LastStableOffset);
-                                rd_kafka_buf_read_i32(rkbuf, &AbortedTxCnt);
-                                /* Ignore aborted transactions for now */
-                                if (AbortedTxCnt > 0)
-                                        rd_kafka_buf_skip(rkbuf,
-                                                          AbortedTxCnt * (8+8));
+                                rd_kafka_buf_read_i32(rkbuf,
+                                                      &AbortedTxnCnt);
+
+                                if (rkb->rkb_rk->rk_conf.isolation_level ==
+                                    RD_KAFKA_READ_COMMITTED)
+                                        end_offset = hdr.LastStableOffset;
+
+                                if (AbortedTxnCnt > 0) {
+                                        size_t aborted_txn_list_idx;
+
+                                        if (unlikely(rkb->rkb_rk->rk_conf.isolation_level ==
+                                            RD_KAFKA_READ_UNCOMMITTED))
+                                                rd_rkb_log(rkb, LOG_WARNING, "FETCH",
+                                                        "%.*s [%"PRId32"]: "
+                                                        "%"PRId32" aborted transaction(s) "
+                                                        "encountered in READ_UNCOMMITTED "
+                                                        "fetch response: ignoring",
```

Comment:
i've now put this check after the aborted transactions are parsed, changed the warning to an error and destroy the aborted transactions data in the if block. reasoning: this case should never be able to happen in practice. if it does (for some reason that should never be able to happen), the user wants uncommitted values, so the aborted transactions should be ignored. since this should never happen, I the inefficiency of reading/parsing aborted transactions when they aren't required is a non issue.